EventFinder-DDLs.sql

```diff
@@ -0,0 +1,190 @@
+CREATE TABLE DoB (
+    DateOfBirth DATE PRIMARY KEY,
+    Age INT NOT NULL
+);
+
+CREATE TABLE User (
+    UserID INT PRIMARY KEY AUTO_INCREMENT,
+    FirstName VARCHAR(50) NOT NULL,
+    LastName VARCHAR(50) NOT NULL,
+    DateOfBirth DATE NOT NULL,
+    Gender VARCHAR(50) NOT NULL,
+    FOREIGN KEY (DateOfBirth) REFERENCES DoB(DateOfBirth) ON DELETE NO ACTION ON UPDATE CASCADE
+);
+
+CREATE TABLE CreditCard (
+    CardNumber VARCHAR(16) PRIMARY KEY,
+    ExpiryDate DATE NOT NULL,
+    HolderName VARCHAR(100) NOT NULL,
+    CVC INT NOT NULL
+);
+
+CREATE TABLE Organizer (
+    OrganizerUserID INT PRIMARY KEY,
+    OrganizerContactInfo VARCHAR(200) NOT NULL,
+	FOREIGN KEY (OrganizerUserID) REFERENCES User(UserID) ON DELETE CASCADE
+);
+
+CREATE TABLE Event (
+    EventID INT PRIMARY KEY AUTO_INCREMENT,
+    Title VARCHAR(200) NOT NULL,
+    StartDate TIMESTAMP NOT NULL,
+    EndDate TIMESTAMP NOT NULL,
+    Description VARCHAR(200),
+    LocationAddress VARCHAR(200),
+    OrganizerUserID INT,
+    FOREIGN KEY (LocationAddress) REFERENCES Location(Address) ON DELETE NO ACTION,
+    FOREIGN KEY (OrganizerUserID) REFERENCES User(UserID) ON DELETE SET NULL,
+    FOREIGN KEY (OrganizerUserID) REFERENCES Organizer(OrganizerUserID) ON DELETE NO ACTION
+);
+
+CREATE TABLE PaidEvent (
+    EventID INT PRIMARY KEY,
+    Price INT NOT NULL,
+    FOREIGN KEY (EventID) REFERENCES Event(EventID) ON DELETE CASCADE
+);
+
+CREATE TABLE Staff (
+    StaffID INT PRIMARY KEY AUTO_INCREMENT,
+    Name VARCHAR(100) NOT NULL,
+    PayRate INTEGER NOT NULL
+);
+
+CREATE TABLE Location (
+    Address VARCHAR(200) PRIMARY KEY,
+    Name VARCHAR(200) NOT NULL
+);

```

Comment:
Move `Location` to come before `Event` otherwise there will be a foreign key error.